File: app/src/main/java/com/nirbhor/app/ui/components/Surfaces.kt
package com.nirbhor.app.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.nirbhor.app.ui.theme.Dimens
import com.nirbhor.app.ui.theme.NirbhorTheme

/**
 * Card shadow (README): calm cards get a soft ink shadow; urgent cards get a stronger amber shadow.
 * Compose collapses the design's two-layer shadow into a single elevation with a tinted spot colour.
 */
fun Modifier.nbCardShadow(shape: RoundedCornerShape, elevated: Boolean = false): Modifier =
    this.shadow(
        elevation = if (elevated) 12.dp else 8.dp,
        shape = shape,
        ambientColor = Color(0x141B2A26),
        spotColor = if (elevated) Color(0x2EC07138) else Color(0x141B2A26),
    )

/** Standard white card. */
@Composable
fun NbCard(
    modifier: Modifier = Modifier,
    padding: Dp = Dimens.cardPadding,
    radius: Dp = Dimens.radiusCard,
    onClick: (() -> Unit)? = null,
    border: BorderStroke? = null,
    background: Color? = null,
    content: @Composable ColumnScope.() -> Unit,
) {
    val colors = NirbhorTheme.colors
    val shape = RoundedCornerShape(radius)
    Column(
        modifier = modifier
            .nbCardShadow(shape)
            .clip(shape)
            .background(background ?: colors.card)
            .then(if (border != null) Modifier.border(border, shape) else Modifier)
            .then(if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier)
            .padding(padding),
        content = content,
    )
}

/** Due-now / urgent card: 2px warm border + amber shadow. */
@Composable
fun UrgentCard(
    modifier: Modifier = Modifier,
    padding: Dp = Dimens.cardPadding,
    radius: Dp = Dimens.radiusCard,
    content: @Composable ColumnScope.() -> Unit,
) {
    val colors = NirbhorTheme.colors
    val shape = RoundedCornerShape(radius)
    Column(
        modifier = modifier
            .nbCardShadow(shape, elevated = true)
            .clip(shape)
            .background(colors.card)
            .border(BorderStroke(2.dp, colors.warm), shape)
            .padding(padding),
        content = content,
    )
}

/** A flat tinted panel (sage / calm-soft / warm-soft info panels). No shadow. */
@Composable
fun TintPanel(
    background: Color,
    modifier: Modifier = Modifier,
    padding: Dp = Dimens.cardPadding,
    radius: Dp = Dimens.radiusCard,
    border: BorderStroke? = null,
    content: @Composable ColumnScope.() -> Unit,
) {
    val shape = RoundedCornerShape(radius)
    Column(
        modifier = modifier
            .clip(shape)
            .background(background)
            .then(if (border != null) Modifier.border(border, shape) else Modifier)
            .padding(padding),
        content = content,
    )
}

File: lib/ui/components/surfaces.dart
import 'package:flutter/material.dart';

import '../../theme/theme.dart';

/// Card shadow (README): calm cards get a soft ink shadow; urgent cards get a stronger amber one.
List<BoxShadow> nbCardShadow({bool elevated = false}) => [
      BoxShadow(
        color: elevated ? const Color(0x2EC07138) : const Color(0x141B2A26),
        blurRadius: elevated ? 20 : 14,
        offset: Offset(0, elevated ? 6 : 4),
      ),
      const BoxShadow(
        color: Color(0x141B2A26),
        blurRadius: 6,
        offset: Offset(0, 1),
      ),
    ];

/// Standard white card.
class NbCard extends StatelessWidget {
  const NbCard({
    super.key,
    required this.child,
    this.padding = Dimens.cardPadding,
    this.radius = Dimens.radiusCard,
    this.onTap,
    this.border,
    this.background,
    this.crossAxisAlignment = CrossAxisAlignment.start,
  });

  final Widget child;
  final double padding;
  final double radius;
  final VoidCallback? onTap;
  final BoxBorder? border;
  final Color? background;
  final CrossAxisAlignment crossAxisAlignment;

  @override
  Widget build(BuildContext context) {
    final shape = BorderRadius.circular(radius);
    final body = Padding(
      padding: EdgeInsets.all(padding),
      child: Column(
        crossAxisAlignment: crossAxisAlignment,
        mainAxisSize: MainAxisSize.min,
        children: [child],
      ),
    );
    return DecoratedBox(
      decoration: BoxDecoration(
        color: background ?? context.colors.card,
        borderRadius: shape,
        border: border,
        boxShadow: nbCardShadow(),
      ),
      child: onTap == null
          ? body
          : Material(
              color: Colors.transparent,
              child: InkWell(borderRadius: shape, onTap: onTap, child: body),
            ),
    );
  }
}

/// Due-now / urgent card: 2px warm border + amber shadow.
class UrgentCard extends StatelessWidget {
  const UrgentCard({
    super.key,
    required this.child,
    this.padding = Dimens.cardPadding,
    this.radius = Dimens.radiusCard,
  });

  final Widget child;
  final double padding;
  final double radius;

  @override
  Widget build(BuildContext context) => DecoratedBox(
        decoration: BoxDecoration(
          color: context.colors.card,
          borderRadius: BorderRadius.circular(radius),
          border: Border.all(color: context.colors.warm, width: 2),
          boxShadow: nbCardShadow(elevated: true),
        ),
        child: Padding(
          padding: EdgeInsets.all(padding),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            mainAxisSize: MainAxisSize.min,
            children: [child],
          ),
        ),
      );
}

/// A flat tinted panel (sage / calm-soft / warm-soft info panels). No shadow.
class TintPanel extends StatelessWidget {
  const TintPanel({
    super.key,
    required this.background,
    required this.child,
    this.padding = Dimens.cardPadding,
    this.radius = Dimens.radiusCard,
    this.border,
  });

  final Color background;
  final Widget child;
  final double padding;
  final double radius;
  final BoxBorder? border;

  @override
  Widget build(BuildContext context) => Container(
        decoration: BoxDecoration(
          color: background,
          borderRadius: BorderRadius.circular(radius),
          border: border,
        ),
        padding: EdgeInsets.all(padding),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          mainAxisSize: MainAxisSize.min,
          children: [child],
        ),
      );
}
